Service/DataService.php

```diff
@@ -966,6 +971,67 @@ public function finalizeDraft(Revision $revision, FormInterface &$form = null, $
         return $revision;
     }
 
+    private function checkIfTranslationExist(Revision $revision, FormInterface $form): bool
+    {
+        $localeFieldName = $revision->getContentType()->getLocaleField();
+        $translationFieldName = $revision->getContentType()->getTranslationField();
+
+        if ($localeFieldName === null || $translationFieldName === null) {
+            return false;
+        }
+
+        if (!isset($form->get('data')->getData()[$localeFieldName]) || !isset($form->get('data')->getData()[$translationFieldName])) {
+            return false;
```

Comment:
You must check the raw data not the field in the form (as those fields might be hidden in tabsFieldType or ContainerFieldType